File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include<algorithm>
using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    	int lotIndex = 0;
	int winIndex = 0;
	int match = 0;
	int blank = 0;
	sort(lottos.begin(), lottos.end());
	sort(win_nums.begin(), win_nums.end());

	while (lotIndex < lottos.size() && winIndex < win_nums.size())
	{
		if (lottos[lotIndex] == 0)
		{
			blank++;
			lotIndex++;
            continue;
		}

		if (lottos[lotIndex] == win_nums[winIndex])
		{
			lotIndex++;
			winIndex++;
			match++;
		}
		else if (lottos[lotIndex] > win_nums[winIndex])
		{
			winIndex++;
		}
		else if (lottos[lotIndex] < win_nums[winIndex])
		{
			lotIndex++;
		}
	}
	int high = match + blank;
	int low = match;

	if (high <= 1)
		answer.push_back(6);
	else if (high >= 2)
		answer.push_back(7 - high);

	if (low <= 1)
		answer.push_back(6);
	else if (low >= 2)
		answer.push_back(7 - low);
    return answer;
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cs
using System;
using System.Linq;
using System.Collections.Generic;
public class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        Dictionary<int,int> dict = new Dictionary<int,int>();
        foreach(var item in lottos)
        {
            if(dict.ContainsKey(item))
            {
                dict[item]++;
            }
            else
            {
                dict.Add(item,1);
            }
        }
        foreach(var item in win_nums)
        {
            if(dict.ContainsKey(item))
            {
                dict[item]++;
            }
            else
            {
                dict.Add(item,1);
            }
        }
        
        int score = 0;
        int cnt = 0;
        foreach(var item in dict)
        {
            if(item.Key == 0)
            {
                cnt = item.Value;
            }
            else if(item.Value == 2) score++;
        }
        int[] answer = new int[2];
        answer[0] = 7- (score+cnt >=2 ? score+cnt : 1);
        answer[1] = 7 - (score >=2 ? score : 1);
        
        return answer;
    }
}